### ISSUE TITLE
Cocoaopengl fails to compile on OSX

### ISSUE THREAD

**Issue by matham** [libsdl-org/SDL#6586]
Upgrading sdl2 to 2.26.0 for conda (https://github.com/conda-forge/sdl2-feedstock/pull/32) resulted in the following error:

```sh
  CC     build/SDL_cocoaopengl.lo
/Users/runner/miniforge3/conda-bld/sdl2_1669148518848/work/src/video/cocoa/SDL_cocoaopengl.m:103:107: error: no visible @interface for 'SDLOpenGLContext' declares the selector 'pixelFormat'
        CVDisplayLinkSetCurrentCGDisplayFromOpenGLContext(self->displayLink, [self CGLContextObj], [[self pixelFormat] CGLPixelFormatObj]);
                                                                                                     ~~~~ ^~~~~~~~~~~
1 error generated.
make: *** [Makefile:694: build/SDL_cocoaopengl.lo] Error 1
```

The change seems to originate here: https://github.com/libsdl-org/SDL/commit/7c760f7f79f65cc628fc43db9d507f051475a43f.

Here's the full log:
[log.txt](https://github.com/libsdl-org/SDL/files/10070953/log.txt)

**Comment by jmroot** [comment]
MacPorts is also seeing this on <= 10.9, though we're also seeing a second error:
```
/opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_devel_libsdl2/libsdl2/work/SDL2-2.26.0/src/video/cocoa/SDL_cocoaopengl.m:91:98: error: expected ')'
        CVDisplayLinkSetOutputCallback(self->displayLink, &DisplayLinkCallback, (__bridge void * _Nullable) self);
                                                                                                 ^
/opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_devel_libsdl2/libsdl2/work/SDL2-2.26.0/src/video/cocoa/SDL_cocoaopengl.m:91:81: note: to match this '('
        CVDisplayLinkSetOutputCallback(self->displayLink, &DisplayLinkCallback, (__bridge void * _Nullable) self);
                                                                                ^
/opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_devel_libsdl2/libsdl2/work/SDL2-2.26.0/src/video/cocoa/SDL_cocoaopengl.m:103:107: error: no visible @interface for 'SDLOpenGLContext' declares the selector 'pixelFormat'
        CVDisplayLinkSetCurrentCGDisplayFromOpenGLContext(self->displayLink, [self CGLContextObj], [[self pixelFormat] CGLPixelFormatObj]);
                                                                                                     ~~~~ ^~~~~~~~~~~
2 errors generated.
```

Full log [here](https://build.macports.org/builders/ports-10.9_x86_64-builder/builds/207518/steps/install-port/logs/stdio).

**Comment by icculus** [comment]
Oh, whoops, `-[NSOpenGLContext pixelFormat]` didn't show up until 10.10. I just assumed it was such a basic obvious interface that it was probably there since 10.0.

I'll fix it.

**Comment by icculus** [comment]
ok, this is fixed in SDL2, SDL3, and the release-2.26.x branch for shipping in 2.26.1

**Comment by jmroot** [comment]
Excellent, thank you.
